### PR TITLE
[IMP] supported_versions: support saas-15.1 and deprecate other SaaS versions

### PR DESCRIPTION
**Pull request by AntoineVDV**

### CODE REVIEW

**Comment by AntoineVDV** [comment]
@robodoo rebase-ff

**Comment by AntoineVDV** [comment]
@jcs-odoo I added some additional details on the upgrade process requested by APR. I didn't make too much effort bringing them up to the written English standards expected for the docs so feel free to checkout the branch, amend the last commit and merge yourself ;)

@robodoo delegate=jcs-odoo

File: content/administration/maintain/supported_versions.rst [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
 Odoo provides support and bug fixing **for the 3 last major versions** of Odoo.
 
-Users hosted on **Odoo Online** may use intermediary versions (sometimes called *SaaS versions*) that are
-supported as well. These versions are not published for Odoo.sh or On-Premise
-installations.
-
+.. note::
+   Odoo also releases intermediary versions (called *SaaS versions*) on the Odoo Online hosting
+   every two months. Users on Odoo Online can then benefit from latest features of Odoo. Only the
+   latest SaaS version is supported. Users hosted on Odoo Online are then invited to upgrade
```
Comment (jcs-odoo) [inline content/administration/maintain/supported_versions.rst:13]: Not sure what 
"Only the latest SaaS version is supported."
means here.

File: content/administration/maintain/supported_versions.rst [hunk 2]
```diff
@@ -61,11 +62,8 @@ This matrix shows the support status of every version.
 
     N/A Never released for this platform
 
-    |orange| Some of our older customers may still run this version on our Odoo Online servers, we provide help only on blocking issues and advise you to upgrade.
-
```
Comment (jcs-odoo) [inline content/administration/maintain/supported_versions.rst:65]: Is the removal of orange to "push" admins to upgrade their old versions?

If I'm correct, databases on really old versions _must_ upgrade soon. Wouldn't we make a distinction on this page and use red for these versions, and orange for everything else? 

(I assume we won't, but I still wanted to share my thoughts :) )

File: content/administration/maintain/online.rst [hunk 3]
```diff
@@ -37,12 +37,29 @@ Upgrade
...
 Upgrade the database to the latest Odoo version.
 
+In your database you will be invited to upgrade if you are not on the latest version we support on
+the SaaS.
+
+- If you are currently on a long-term supported version of Odoo
+
+  - The upgrade button will propose an upgrade to the latest version (e.g: 13 to 15).
+  - If you are already on the latest version, you will be invited to upgrade to the last Online
+    Version but you can dismiss the request and stay on the latest supported version. You won’t
```
Comment (jcs-odoo) [inline content/administration/maintain/online.rst:47]: Is "Online Version" what's defined as "SaaS Version" in the supported versions page?
I'll change it to "SaaS version" to stay consistent, but let me know if I misunderstood, @apr-odoo

**Review by jcs-odoo — COMMENTED** [review]
I'll force push some changes, then.

cc @Donapi . it may affect your upgrade PR, by the way.

**Pull request closed without merging**